webapp/channels/src/actions/views/channel.ts

```diff
@@ -118,8 +121,7 @@ export function switchToChannel(channel: Channel & {userId?: string}): ActionFun
             }
             getHistory().push(`${teamUrl}/messages/@${channel.name}`);
         } else if (channel.type === Constants.GM_CHANNEL) {
-            const gmChannel = getChannel(state, channel.id);
-            getHistory().push(`${teamUrl}/channels/${gmChannel.name}`);
+            getHistory().push(`${teamUrl}/channels/${channel.name}`);
```

Comment:
Actually, we need to revert this because there's some ugly stuff here which is hidden by the TS types being wrong. The quick switch modal passes something which isn't quite a real channel object in certain cases, and for GM channels, the `channel.name` is overwritten to be the display name so this doesn't work.

The `SwitchChannelProvider` has overwritten the name of the channel ever since GM channels were first added (https://github.com/mattermost/mattermost-webapp/commit/12044aa0a61dbfc26fddca66a2802ee3190a7155#diff-7b6d2a0dcf7ccfbfbabb1b46a635eb5eb7b7197d5a1a7cc5d7a4f478cb199501R82) for some reason. I think that goes back to the quick switch modal autocompleting channels based on their `name` originally, not their display name. Unfortunately, that means that the name gets mangled by the time the `QuickSwitchModal` passes it to this action, so https://github.com/mattermost/mattermost-webapp/pull/2752 refetched the channel to get around that since we only overwrite it for GM channels.

Ideally, we wouldn't need any of this because the `SwitchChannelProvider` wouldn't mess with the channel object at all, but let's not go down that rabbit hole in this PR

Reply:
Reverted, and made it a bit safer.